sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -306,13 +306,24 @@ namespace Azure { namespace Core { namespace Http {
         NextHttpPolicy nextHttpPolicy) const override;
   };
 
+  /**
+   * @brief Defines options for getting token.
+   */
+  struct GetTokenOptions
```

Comment:
```suggestion
  struct TokenRequestOptions
```

Note that other languages call this struct "TokenRequestContext".  I don't like "Context", though we may want to match that name to match what other language SDKs are doing

https://docs.microsoft.com/en-us/dotnet/api/azure.core.tokenrequestcontext?view=azure-dotnet
https://docs.microsoft.com/en-us/java/api/com.azure.identity.chainedtokencredential.gettoken?view=azure-java-stable

Reply:
I actually like the "TokenRequestContext/CancellationToken", and I like "TokenRequestOptions", but this is what was decided on during the review meeting.
I would also be ok with "BearerTokenAuthenticationPolicyOptions", and I am not a fan of a verb in the class name. But also we had that time to discuss, all of us were here, and we decided on GetTokenOptions, that the only thing that makes me hesitate.

@JeffreyRichter, @schaabs, please let me know if you would still prefer GetTokenOptions name instead of TokenRequestOptions.